### DISCUSSION TITLE
Feature: Identify Newly Flagged KRIs

### DISCUSSION THREAD

**Discussion by mattroumaya** [Gilead-Public/gsm#1693] [Ideas]
## Feature Details
- When longitudinal data is available, identify sites with new red flags (+/- 2)
- Add boxes to existing report to show # of newly flagged sites per KRI
- Surface KRI history in study overview table -- display change from previous snapshot regardless of current KRI status.

Boxes should have:
-  n new red flags
-  n total red flags
-  n total amber flags
- some indication for each site that is newly flagged
**Likely to be done as part of update in `{rbm-viz}`**

## additional supplemental table at top of report
- just shows trends
- site number, basic info, filterable version of `{grail}` table
  - filter on: current KRI value, previous KRI value
  - e.g., prev KRI value = 0, current KRI value = +2, only show sites with that criteria

**Comment by mattroumaya** [comment]
To do:
- Do a mock-up to show how the report would look with this implementation
- Linked to #1050

**Comment by mattroumaya** [comment]
link to rbm-viz issue

**Comment by mattroumaya** [comment]
@jwildfire to add details

**Comment by jwildfire** [comment]
Here are my current thoughts about how to incorporate information about newly flagged KRIs in our standard KRI report in {gsm}. I think this data is all fairly easily available in the v1.9 data model. Once we agree on details, this should almost definitely be broken out in to multiple issues. 
 
**Overview Section**
- ** Show “XX Newly Flagged KRIs” next to “XX Red/Amber KRIs”
- ! Overview table - Add “Sites with New Flags” as option to filter 
- ! Overview table - Invert color scheme (Change background color to red/amber, with white icon) for newly flagged KRIs 
- ** Add a new listing of Newly flagged KRIs
    - Columns = Site, KRI, current_metric, current_score, current_flag, previous_metric, previous_score, previous_flag

**KRI charts/table**
- ! Scatter plot - use fill color to indicate newly flagged KRIs. Filled = new flag. Empty = no change. 
- ** Add Prev Flag to summary table
- Bar Charts - Add a line showing the change from the previous assessment
    - Probably overkill and don’t want to do this
- ** Add new heatmap grid showing all flag values. Columns = snapshots. Rows = sites. Fill color = flag status. 
    - Include filter for “Sites with Red/Amber” and all sites
    - Add a new chart tab for this
- ** Summary table. Add column for previous flag

! = rbm-viz update
** = target for v1.10

**Comment by jwildfire** [comment]
cc: @Craftech-RR
